### ISSUE TITLE
Modules with no outputs have no exported `for_each` keys

### ISSUE THREAD

**Issue by disog** [hashicorp/terraform#33682]
### Terraform Version

```shell
Terraform v1.5.5
on windows_amd64
```

### Terraform Configuration Files

https://github.com/diegosogari/terraform-issue

### Debug Output

https://gist.github.com/diegosogari/6a3e0bb9df3cdb298da7351ebd38b4a4

### Expected Behavior

The value of `no_output` should be identical to that of `with_output`.

### Actual Behavior

The value of `no_output` is different from that of `with_output`.

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### Additional Context

There's nothing atypical about my situation.

**Comment by carlosjgp** [comment]
I'm seeing the same error with
```
Terraform v1.5.3
on linux_amd64
```

**Comment by jbardin** [comment]
Closed via #34545
